Building R from source on Ubuntu in WSL 2 fails

# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

<!-- To help us debug your issue, please complete these sections: -->

## What you were trying to do (and why)

I was trying to install R from source in Ubuntu 20.10 running on WSL 2. The output of `brew gist-logs r` [can be found here](https://gist.github.com/eced8272c083a7e6bff48bd1ea602b8d).

Note that I couldn't quite fix all `brew doctor` issues as `gcc` depends on `isl@0.18`, which is deprecated.

## What happened (include command output)

The build failed with the message that `rpc/types.h` could not be found.

<details>
  <summary>Command output</summary>
  <pre>

  ==> Downloading https://cran.r-project.org/src/base/R-4/R-4.0.3.tar.gz
Already downloaded: /home/vsong/.cache/Homebrew/downloads/d0aba331f24b5d794d1b83d73e826f2ddaeb191efa58fd088d80eaba45d7450b--R-4.0.3.tar.gz
==> Reinstalling r
==> ./configure --prefix=/home/linuxbrew/.linuxbrew/Cellar/r/4.0.3 --enable-memory-profiling --with-lapack --enable-R-shlib --disable-java --with-blas=-L/home/linuxbrew/.linuxbrew/opt/openblas/lib -lopenblas SED=/usr/bin/sed --libdir=/home/linuxbrew/.
==> make
Last 15 lines from /home/vsong/.cache/Homebrew/Logs/r/02.make:
gcc-5 -I../../src/extra -I/usr/include/tirpc -I. -I../../src/include -I../../src/include -I/home/linuxbrew/.linuxbrew/Cellar/curl/7.74.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/opt/readline/include -I../../src/nmath -DHAVE_CONFIG_H   -fopenmp -fpic  -g -O2  -c serialize.c -o serialize.o
gcc-5 -I../../src/extra -I/usr/include/tirpc -I. -I../../src/include -I../../src/include -I/home/linuxbrew/.linuxbrew/Cellar/curl/7.74.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/opt/readline/include -I../../src/nmath -DHAVE_CONFIG_H   -fopenmp -fpic  -g -O2  -c sort.c -o sort.o
saveload.c:115:23: fatal error: rpc/types.h: No such file or directory
compilation terminated.
make[3]: *** [../../Makeconf:125: saveload.o] Error 1
make[3]: *** Waiting for unfinished jobs....
serialize.c:885:23: fatal error: rpc/types.h: No such file or directory
compilation terminated.
make[3]: *** [../../Makeconf:125: serialize.o] Error 1
make[3]: Leaving directory '/tmp/r-20201210-10532-1h907y5/R-4.0.3/src/main'
make[2]: *** [Makefile:138: R] Error 2
make[2]: Leaving directory '/tmp/r-20201210-10532-1h907y5/R-4.0.3/src/main'
make[1]: *** [Makefile:28: R] Error 1
make[1]: Leaving directory '/tmp/r-20201210-10532-1h907y5/R-4.0.3/src'
make: *** [Makefile:61: R] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting

  </pre>
</details>

## What you expected to happen

The build should have succeeded.

## Step-by-step reproduction instructions (by running `brew install` commands)

`brew install -s r`

--- a/Formula/r.rb
+++ b/Formula/r.rb
@@ -4,7 +4,7 @@ class R < Formula
   url "https://cran.r-project.org/src/base/R-4/R-4.0.3.tar.gz"
   sha256 "09983a8a78d5fb6bc45d27b1c55f9ba5265f78fa54a55c13ae691f87c5bb9e0d"
   license "GPL-2.0-or-later"
-  revision 2
+  revision OS.mac? ? 1 : 2
 
   livecheck do
     url "https://cran.rstudio.com/banner.shtml"
@@ -15,7 +15,6 @@ class R < Formula
     sha256 "01bf5a586b3699a50b5e7ec6b7eebab213ef47dfa94754db497d0f41ac5a6110" => :big_sur
     sha256 "c600906bfe86b80c5c14129148692364a6a5c2d4c1417e7c9f8a4eff5f508ec2" => :catalina
     sha256 "22731d36544e6eb86b88e89638225133c8975377c3c4cdd1cc06e61e72f846b9" => :mojave
-    sha256 "0429f284050a1f1fde4c962cca7fde184b7334cb21fddde287eb38bb16d2c9c8" => :x86_64_linux
   end
 
   depends_on "pkg-config" => :build

--- a/Formula/r.rb
+++ b/Formula/r.rb
@@ -4,7 +4,7 @@ class R < Formula
   url "https://cran.r-project.org/src/base/R-4/R-4.0.3.tar.gz"
   sha256 "09983a8a78d5fb6bc45d27b1c55f9ba5265f78fa54a55c13ae691f87c5bb9e0d"
   license "GPL-2.0-or-later"
-  revision 1
+  revision 2
 
   livecheck do
     url "https://cran.rstudio.com/banner.shtml"
@@ -35,6 +35,7 @@ class R < Formula
     depends_on "libice"
     depends_on "libx11"
     depends_on "libxt"
+    depends_on "libtirpc"
   end
 
   # needed to preserve executable permissions on files without shebangs
@@ -82,6 +83,11 @@ class R < Formula
 
     # Avoid references to homebrew shims
     args << "LD=ld" unless OS.mac?
+
+    unless OS.mac?
+      ENV.append "CPPFLAGS", "-I#{Formula["libtirpc"].opt_include}/tirpc"
+      ENV.append "LDFLAGS", "-L#{Formula["libtirpc"].opt_lib}"
+    end
 
     system "./configure", *args
     system "make"

